parrot/src/pfs_table.cc

```diff
@@ -1366,8 +1378,11 @@ int pfs_table::unlink( const char *n )
 {
 	pfs_name pname;
 	int result = -1;
+	char dirname[PFS_PATH_MAX];
+
+	path_dirname(n,dirname);
 
-	if(resolve_name(0,n,&pname,false)) {
+	if(resolve_name(0,dirname,&pname,X_OK|W_OK,false) && resolve_name(0,n,&pname,F_OK,false)) {
```

Comment:
The two checks here are for these example situations:

````
/a/b R_OK
````

`unlink("/a/b/c") = -1 (EPERM)`

and

````
/a/b/c /etc/services R_OK
````

`unlink("/a/b/c") = 0`

Yes?